mr/src/main/java/org/elasticsearch/hadoop/serialization/bulk/DeleteBulkFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.serialization.bulk;
+
+import org.apache.hadoop.io.*;
+import org.elasticsearch.hadoop.cfg.ConfigurationOptions;
+import org.elasticsearch.hadoop.cfg.Settings;
+import org.elasticsearch.hadoop.serialization.Generator;
+import org.elasticsearch.hadoop.serialization.builder.ValueWriter;
+import org.elasticsearch.hadoop.util.EsMajorVersion;
+import org.elasticsearch.hadoop.util.StringUtils;
+
+import java.util.List;
+
+public class DeleteBulkFactory extends AbstractBulkFactory {
+
+    public static final class NoDataWriter implements ValueWriter<Object> {
+
+        @Override
+        public Result write(Object writable, Generator generator) {
```

Comment:
It looks like it only supports Hadoop Writable objects here. For instance, Spark uses regular Java and Scala objects for data types. 

Since integrations can be added that have different types than described here, it makes me wonder if this is really the right place to have the connector ignore the document content for the bulk request. I don't think a ValueWriter will have enough wisdom at runtime to make the decision on a call-by-call basis. Maybe there needs to be work done in order to have a new `BulkCommand` that is returned from the `DeleteBulkFactory` ignore the document instead.